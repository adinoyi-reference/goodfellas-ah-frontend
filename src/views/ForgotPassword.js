import React, { Component } from 'react';
import { connect } from 'react-redux';
import { forgotPassword } from '../actions/authActions';
import Submitbtn from '../components/shared/Submitbtn';

class ForgotPassword extends Component {
  handleForgotPassword = (e) => {
    e.preventDefault();
    const userEmail = {
      email: e.target.email.value,
    };
    this.props.forgotPassword(userEmail);
  };

  render() {
    const { successMessage, errorMessage } = this.props.auth;
    return (
      <div className="pwreset">
        <div className="container">
          <div className="row">
            <div className="col-md-6 m-auto">
              <p className="lead text-center">Reset password</p>
              <form onSubmit={this.handleForgotPassword}>
                <div className="form-group">
                  <input type="email" className="form-control form-control-lg" placeholder="Email Address" name="email" />
                </div>
                {successMessage ? (
                  <div className="successmsg">
                    <p>{successMessage}</p>
                  </div>
                ) : (
                  <div className="errormsg"><p>{errorMessage}</p></div>
                )}
                <Submitbtn />
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
const mapStateToProps = (state) => ({
  auth: state.auth,
});

export default connect(
  mapStateToProps,
  { forgotPassword },
)(ForgotPassword);
