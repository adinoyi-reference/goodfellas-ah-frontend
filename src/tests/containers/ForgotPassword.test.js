import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ForgotPassword from '../../containers/ForgotPassword';
import Root from '../../root';

let wrapped;

beforeEach(() => {
  wrapped = mount(
    <Root>
      <ForgotPassword />
    </Root>,
  );
});

afterEach(() => wrapped.unmount());
describe('ForgotPassword UI', () => {
  describe('render', () => {
    test('container should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });
    it('has two input types', () => {
      expect(wrapped.find('input').length).toEqual(2);
    });
  });
});
