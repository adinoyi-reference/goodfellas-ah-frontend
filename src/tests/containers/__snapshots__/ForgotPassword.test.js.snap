// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword render container should render as expected 1`] = `
<_default>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(ForgotPassword)>
      <ForgotPassword
        forgotPassword={[Function]}
      >
        <form
          onSubmit={[Function]}
        >
          <InputField
            name="email"
            placeholder="Email Address"
            type="email"
          >
            <div
              className="form-group"
            >
              <input
                className="form-control form-control-lg"
                name="email"
                placeholder="Email Address"
                type="email"
              />
            </div>
          </InputField>
          <Submitbtn>
            <div>
              <input
                className="btn btn-block mt-4 resetbtn"
                type="submit"
              />
            </div>
          </Submitbtn>
        </form>
      </ForgotPassword>
    </Connect(ForgotPassword)>
  </Provider>
</_default>
`;
